src/maps/mapView/mapView.tsx

```diff
@@ -0,0 +1,149 @@
+import { StyleSheet, View, findNodeHandle } from 'react-native';
+import {
+  NavViewManager,
+  sendCommand,
+  commands,
+  type LatLng,
+} from '../../shared';
+import {
+  getMapViewController,
+  type Circle,
+  type GroundOverlay,
+  type MapViewProps,
+  type Marker,
+  type Polygon,
+  type Polyline,
+} from '..';
+
+export const MapView = (props: MapViewProps) => {
+  const mapViewRef = useRef<any>(null);
+  const [viewId, setViewId] = useState<number | null>(null);
+
+  const { onMapViewControllerCreated } = props;
+
+  /**
+   * @param {any} _ref - The reference to the NavViewManager component.
+   */
+  const onRefAssign = (_ref: any) => {
+    if (mapViewRef.current !== _ref) {
+      mapViewRef.current = _ref;
+    }
+  };
+
+  useEffect(() => {
+    if (!mapViewRef.current) {
+      return;
+    }
+    const _viewId = findNodeHandle(mapViewRef.current) || 0;
+    if (viewId !== _viewId) {
+      setViewId(_viewId);
+
+      const isNavigationEnabled = false;
+
+      const stylingOptions = {};
+
+      const args = [stylingOptions, isNavigationEnabled];
+
+      setTimeout(() => {
+        sendCommand(_viewId, commands.createFragment, args);
+      });
+
+      onMapViewControllerCreated(getMapViewController(_viewId));
+    }
+  }, [onMapViewControllerCreated, viewId]);
+
+  const onMapClick = useCallback(
+    ({ nativeEvent: latlng }: { nativeEvent: LatLng }) => {
+      props.mapViewCallbacks?.onMapClick?.(latlng);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onMapReady = useCallback(() => {
+    props.mapViewCallbacks?.onMapReady?.();
+  }, [props.mapViewCallbacks]);
+
+  const onMarkerClick = useCallback(
+    ({ nativeEvent: marker }: { nativeEvent: Marker }) => {
+      props.mapViewCallbacks?.onMarkerClick?.(marker);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onPolylineClick = useCallback(
+    ({ nativeEvent: polyline }: { nativeEvent: Polyline }) => {
+      props.mapViewCallbacks?.onPolylineClick?.(polyline);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onPolygonClick = useCallback(
+    ({ nativeEvent: polygon }: { nativeEvent: Polygon }) => {
+      props.mapViewCallbacks?.onPolygonClick?.(polygon);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onCircleClick = useCallback(
+    ({ nativeEvent: circle }: { nativeEvent: Circle }) => {
+      props.mapViewCallbacks?.onCircleClick?.(circle);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onGroundOverlayClick = useCallback(
+    ({ nativeEvent: groundOverlay }: { nativeEvent: GroundOverlay }) => {
+      props.mapViewCallbacks?.onGroundOverlayClick?.(groundOverlay);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  const onMarkerInfoWindowTapped = useCallback(
+    ({ nativeEvent: marker }: { nativeEvent: Marker }) => {
+      props.mapViewCallbacks?.onMarkerInfoWindowTapped?.(marker);
+    },
+    [props.mapViewCallbacks]
+  );
+
+  return (
+    <View style={props.style ?? styles.defaultStyle}>
+      <NavViewManager
```

Comment:
Is NavViewManager still good name for handling both MapViews and NavigationViews?

Reply:
`MapViewManager`?
`ViewManager`?

Reply:
Had a discussion about this and decided to keep the naming as is.